### DISCUSSION TITLE
How does Pull work using GIT Desktop

### DISCUSSION THREAD

**Discussion by TG-009** [community/community#22986] [New to GitHub]
<p>Okay, new person here.</p>
<p>I am using GitHub Desktop, there are 2 people on our team.</p>
<p>I will call the other person “OtherPerson”.</p>
<p>A repo has been created on GitHub and the OtherPerson has created a branch called “Branch_Aug_20” from the master and has worked on changes locally (on the OtherPerson local computer) for a while and now pushes that branch to the remote repo on GitHub.</p>
<p>I log into GitHub desktop and want to “Pull” the branch “Branch_Aug_20” to my local computer.</p>
<p>I choose the “Current Repository” in the top left corner to be the repo that we are both working on.</p>
<p>Question:  Do I need to create a branch myself before I can “Pull” the branch “Branch_Aug_20”?</p>
<p>If I don’t create any branch myself and just “Pull”, does a branch get created automatically and what will it be called?</p>
<p>OR</p>
<p>Will I get a local branch called “Branch_Aug_20”?</p>
<p>Question:  If I first create a branch called “MyBranch1”, and then “Pull” “Branch_Aug_20”, does that “Pull” get associated with my branch called “MyBranch1”?</p>
<p>I don’t want to screw up the files by trying a bunch of things when I am not sure what will happen.</p>
<p>Thanks for any replies.</p>

**Comment by YoricW** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/tg-009">@tg-009</a> wrote:</p>
</blockquote>
<blockquote>
<p>Question:  Do I need to create a branch myself before I can “Pull” the branch “Branch_Aug_20”?</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>If I don’t create any branch myself and just “Pull”, does a branch get created automatically and what will it be called?</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>OR</p>
</blockquote>
<blockquote>
<p>Will I get a local branch called “Branch_Aug_20”?</p>
</blockquote>
<p>You just select branch “Branch_Aug_20” and pull,<br>
It will create “Branch_Aug_20” locally.</p>
<blockquote>
<p>Question:  If I first create a branch called “MyBranch1”, and then “Pull” “Branch_Aug_20”, does that “Pull” get associated with my branch called “MyBranch1”?</p>
</blockquote>
<p>It will not be associated.<br>
If you create branch “MyBranch1” it will be a new unrelated branch.</p>

**Comment by TG-009** [comment]
<p>Thank you for the answers.  It seems so obvious now that I think about it and see it in action.</p>
